### PR TITLE
GCP CAR connector

### PR DESCRIPTION
**Pull request by thummalar**
GCP CAR connector for Compute Engine VM instances

### CODE REVIEW

File: connectors/gcp/app.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+version = '1.0.0'
+
+
+class App(BaseApp):
+
+    def __init__(self):
+        super().__init__('This script is used for pushing asset data to CP4S CAR ingestion microservice')
+        # Parameters need to connect data source
+        self.parser.add_argument('-client_email', dest='client_email', default=os.getenv('CONFIGURATION_AUTH_CLIENT_EMAIL', None),
```
Comment (omkar-g) [inline connectors/gcp/app.py]: We'll soon have update in framework where we want `dest` same as env variable for example `CONFIGURATION_AUTH_CLIENT_EMAIL`

File: connectors/gcp/configurations/config.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                "min": 1,
+                "max": 1000
+            }
+        }
+    },
+    "configuration": {
+        "auth": {
+            "type": "fields",
+            "client_email": {
+                "type": "password"
```
Comment (omkar-g) [inline connectors/gcp/configurations/config.json:15]: Why is email id set as password?

File: connectors/gcp/configurations/lang_en.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+    "connection": {
+        "selfSignedCert": {
```
Comment (omkar-g) [inline connectors/gcp/configurations/lang_en.json]: If its a private key lets name this variable as private key.

File: connectors/gcp/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+car-connector-framework==3.0.1
```
Comment (omkar-g) [inline connectors/gcp/requirements.txt]: We have a new framework release 3.0.2. It shouldn't need any changes in connector.

File: connectors/gcp/configurations/lang_en.json (hunk 3 above)
Comment (omkar-g) [reply connectors/gcp/configurations/lang_en.json]: And move it with client email

File: connectors/gcp/app.py (hunk 1 above)
Comment (thummalar) [reply connectors/gcp/app.py]: We have updated the 'dest' same as env variable. we will publish this code changes with GKE code changes.

File: connectors/gcp/requirements.txt (hunk 4 above)
Comment (thummalar) [reply connectors/gcp/requirements.txt]: We will remove car-connector-framework module from requirements file

File: connectors/gcp/configurations/lang_en.json (hunk 3 above)
Comment (thummalar) [reply connectors/gcp/configurations/lang_en.json]: We are keeping configuration similar to Google Chronical UDI connector. In UDI connector Private key is mapped to config variable 'selfSignedCert'.

File: connectors/gcp/configurations/config.json (hunk 2 above)
Comment (thummalar) [reply connectors/gcp/configurations/config.json:15]: We are keeping configuration similar to Google Chronical UDI connector. In UDI connector email type is password.

File: connectors/gcp/app.py (hunk 1 above)
Comment (thummalar) [reply connectors/gcp/app.py]: 'dest' is updated as same as env variables 'CONNECTION_SELFSIGNEDCERT', 'CONFIGURATION_AUTH_CLIENT_EMAIL'

File: connectors/gcp/requirements.txt (hunk 4 above)
Comment (thummalar) [reply connectors/gcp/requirements.txt]: car-connector-framework module is removed from requirements file

File: connectors/gcp/configurations/config.json (hunk 2 above)
Comment (thummalar) [reply connectors/gcp/configurations/config.json:15]: GCP CAR connector configuration kept similar to GCP chronical UDI connector, both connectors using Service Account key for authentication. If both connectors are using same type of authentication, the configuration should be similar.

File: connectors/gcp/configurations/lang_en.json (hunk 3 above)
Comment (thummalar) [reply connectors/gcp/configurations/lang_en.json]: GCP CAR connector configuration is kept similar to GCP chronical UDI connector, both connectors are using Service Account key for authentication. If we use Private key as variable name, the text box is appearing as small and key is large. So it is suggested to use 'selfSignedCert' for larger text box and name the label as 'Private Key' while reviewing PR.

File: connectors/gcp/configurations/config.json (hunk 2 above)
Comment (thummalar) [reply connectors/gcp/configurations/config.json:15]: Updated 'client_email' id type as text

File: connectors/gcp/configurations/lang_en.json (hunk 3 above)
Comment (thummalar) [reply connectors/gcp/configurations/lang_en.json]: Renamed 'selfSignedCert' as 'private_key' and moved to configuration auth section

File: connectors/gcp/requirements.txt (hunk 4 above)
Comment (thummalar) [reply connectors/gcp/requirements.txt]: Updated requirements file with car-connector-framework version 3.0.2

**Review by omkar-g — APPROVED** [review]

**Pull request merged**